src/Components/Forms/test/FieldIdentifierTest.cs

```diff
@@ -183,6 +183,15 @@ public void CanCreateFromExpression_MemberOfChildObject()
         Assert.Equal(nameof(TestModel.StringField), fieldIdentifier.FieldName);
     }
 
+    [Fact]
+    public void CanCreateFromExpression_MemberWithIndexedCollectionEntry()
+    {
+        var model = new TestModel { IndexProperty = new() { false, true } };
+        var fieldIdentifier = FieldIdentifier.Create(() => model.IndexProperty[1]);
+        Assert.Same(model.IndexProperty, fieldIdentifier.Model);
+        Assert.Equal("[1]", fieldIdentifier.FieldName);
+    }
```

Comment:
Could you also add test cases where the key values are other types? For example, string, `System.Object`, null.

Reply:
Also it would need to distinguish between keys with different values but identical string representations, for example the int `1` and the string `1` (which could be used to look up distinct values in a `Dictionary<object, object>`).